lib/src/main/java/com/diffplug/spotless/json/JsonFormatterStep.java

```
@@ -0,0 +1,111 @@
+	}
+
+	private static final class State implements Serializable {
+		private static final long serialVersionUID = 1L;
+
+		private final int indentSpaces;
+		private final boolean useTabs;
+		private final JarState jarState;
+
+		private State(Integer indent, Provisioner provisioner) throws IOException {
+			if (indent != null) {
+				this.indentSpaces = indent;
+				this.useTabs = false;
+			} else {
+				this.indentSpaces = 20; // it's unlikely that someone will use this many spaces, so should be safe
+				useTabs = true;
+			}
+
+			this.jarState = JarState.from(MAVEN_COORDINATE + DEFAULT_VERSION, provisioner);
+		}
+
+		FormatterFunc toFormatter() {
+			Method objectToString;
+			Method arrayToString;
+			Constructor<?> objectConstructor;
+			Constructor<?> arrayConstructor;
+			try {
+				ClassLoader classLoader = jarState.getClassLoader();
+				Class<?> jsonObject = classLoader.loadClass("org.json.JSONObject");
+				Class<?>[] constructorArguments = new Class[]{String.class};
+				objectConstructor = jsonObject.getConstructor(constructorArguments);
+				objectToString = jsonObject.getMethod("toString", int.class);
+
+				Class<?> jsonArray = classLoader.loadClass("org.json.JSONArray");
+				arrayConstructor = jsonArray.getConstructor(constructorArguments);
+				arrayToString = jsonArray.getMethod("toString", int.class);
+			} catch (ClassNotFoundException | NoSuchMethodException e) {
+				throw new IllegalStateException("There was a problem preparing org.json dependencies", e);
+			}
+
+			return s -> {
+				String prettyPrinted = null;
+				if (s.isEmpty()) {
+					prettyPrinted = s;
+				}
+				if (s.startsWith("{")) {
+					try {
+						Object parsed = objectConstructor.newInstance(s);
+						prettyPrinted = objectToString.invoke(parsed, indentSpaces) + "\n";
+					} catch (InvocationTargetException ignored) {
+						// ignore if we cannot convert to JSON string
```

Swallowing exceptions always leads to pain. Remove the try/catch.

---

Although it's swallowed, we still throw an `AssertionError` to show there's nothing formatted - would you prefer it throws the original `InvocationTargetException`?